File: lab2_linguistics/R/load.R
#' Load in question and answer key
#'
#' @param path the path to the data
#'
#' @returns A data frame containing the question and answer key
load_q_and_a_key <- function(path = here::here("data")) {
  data.table::fread(
    file.path(path, "q_and_a_key.csv")
  ) |>
    tibble::as_tibble() |>
    dplyr::mutate(
      question = as.factor(question)
    )
}


#' Load in lingusitics data
#'
#' @param path the path to the data
#'
#' @returns A data frame containing the linguistics survey data
load_ling_data <- function(path = here::here("data")) {
  # TO FILL IN
}


#' Load in ZIP locations data
#'
#' @param path the path to the data
#'
#' @returns A data frame containing the ZIP locations data
load_zip_data <- function(path = here::here("data")) {
  data.table::fread(
    file.path(path, "zip_locations.csv"),
    colClasses = list(character = c("ZIP"))
  ) |>
    tibble::as_tibble()
}

File: lab2_linguistics/python/load.py
import pandas as pd


def load_q_and_a_key(path = "../data"):
    """
    Load the question and answer key from the given path.
    """
    # Load the question and answer key
    q_and_a_key = pd.read_csv(f"{path}/q_and_a_key.csv")
    return q_and_a_key


def load_ling_data(path = "../data"):
    """
    Load the linguistic data from the given path.
    """
    # FILL IN: Load the linguistic data
    return


def load_zip_data(path = "../data"):
    """
    Load the zip locations data from the given path.
    """
    # Load the zip data
    zip_data = pd.read_csv(
        f"{path}/zip_locations.csv",
        dtype={"ZIP": str}
    )
    return zip_data